cloudevents/generic/options/mqtt/sourceoptions.go

```diff
@@ -18,14 +18,16 @@ type mqttSourceOptions struct {
 	MQTTOptions
 	errorChan chan error
 	sourceID  string
+	clientID  string
 }
 
-func NewSourceOptions(mqttOptions *MQTTOptions, sourceID string) *options.CloudEventsSourceOptions {
+func NewSourceOptions(mqttOptions *MQTTOptions, clientID, sourceID string) *options.CloudEventsSourceOptions {
 	return &options.CloudEventsSourceOptions{
 		CloudEventsOptions: &mqttSourceOptions{
 			MQTTOptions: *mqttOptions,
 			errorChan:   make(chan error),
 			sourceID:    sourceID,
```

Comment:
why do we still need source id here, or the one in line 32?

Reply:
reorg the code, make this more clear